Update User-Forum link (#1641)
openmc-dev/openmc: docs/source/usersguide/troubleshoot.rst

Now when you re-run your problem, it should report exactly where the program
failed. If after reading the debug output, you are still unsure why the program

> failed, send an email to the OpenMC User's Group `OpenMC Discourse Forum <https://openmc.discourse.group/>`_.

Reviewer: ```suggestion
failed, post a message on the `OpenMC Discourse Forum
<https://openmc.discourse.group/>`_.
```
